src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -270,76 +236,32 @@ internal static string GetRegistryVersionKeyForSnapinDiscovery(string majorVersi
             return null;
         }
 
-        internal static string FeatureVersionString
-        {
-            get
-            {
-                return string.Format(System.Globalization.CultureInfo.InvariantCulture, "{0}.{1}", PSVersionInfo.PSVersion.Major, PSVersionInfo.PSVersion.Minor);
-            }
-        }
-
         internal static bool IsValidPSVersion(Version version)
         {
-            if (version.Major == s_psSemVersion.Major)
-            {
-                return version.Minor == s_psSemVersion.Minor;
-            }
-
-            if (version.Major == s_psV6Version.Major)
-            {
-                return version.Minor == s_psV6Version.Minor;
-            }
-
-            if (version.Major == s_psV5Version.Major)
+            if (version is null)
             {
-                return (version.Minor == s_psV5Version.Minor || version.Minor == s_psV51Version.Minor);
+                return false;
             }
 
-            if (version.Major == s_psV4Version.Major)
-            {
-                return (version.Minor == s_psV4Version.Minor);
-            }
-            else if (version.Major == s_psV3Version.Major)
-            {
-                return version.Minor == s_psV3Version.Minor;
-            }
-            else if (version.Major == s_psV2Version.Major)
-            {
-                return version.Minor == s_psV2Version.Minor;
-            }
-            else if (version.Major == s_psV1Version.Major)
+            int minor = version.Minor;
+            switch (version.Major)
             {
-                return version.Minor == s_psV1Version.Minor;
+                case 1:
+                case 2:
+                case 3:
+                case 4:
+                    return minor == 0;
+                case 5:
+                    return minor == 0 || minor == 1;
+                case 6:
+                    return minor >= 0 && minor <= 2;
+                case 7:
+                    return minor >= 0 && minor <= s_psVersion.Minor;
```

Comment:
nit: We can replace this with const later with SG.

Reply:
You mean replacing `s_psVersion.Minor` with a const int? It's doable, but I don't think it's needed.

Reply:
We will have this constant in this file, so it doesn't cost us anything to use it. It will even add consistency to it.

Reply:
That's true too.
We can change it to `Version_Minor` after both PRs are merged (or when resolving conflicts for the source generator PR if there is any).